torch/testing/_internal/common_methods_invocations.py

```diff
@@ -74,30 +75,69 @@ def __init__(self, cls_name=None, test_name=None, *,
                          test_name=test_name, device_type=device_type, dtypes=dtypes,
                          active_if=active_if)
 
+
 class SampleInput(object):
     """Represents sample inputs to a function."""
 
-    # output_process_fn_grad is a function that modifies the output of op compatible with input
-    __slots__ = ['input', 'args', 'kwargs', 'output_process_fn_grad']
+    __slots__ = ['input', 'args', 'kwargs', 'output_process_fn_grad', 'ref', 'extra']
```

Comment:
These OpInfo changes are cool. We should separate them from this PR, however, and add a test to test_ops.py that actually runs the "forward" sample inputs and validates them against the reference.

"Forward" sample inputs present an issue because the current "SampleInputs" are designed to be small and suitable for time consuming tests like the gradgradcheck. We should probably add a new attribute, "forward_samples," that's designed for this reference checking.

Another challenge with creating the forward test is picking ops that are good candidates for it and stress all three types of test (fixture/golden value, reference callable, exception). The type of function I'd like to see use this feature is test_reference_numerics, but that's super tricky because it has all this special logic for conversions and error messages:

https://github.com/pytorch/pytorch/blob/eec48303c0d847d3a75edbc455f5cd1d7ef098b9/test/test_unary_ufuncs.py#L276

I wonder if that kind of additional logic is more the norm than the exception. If so, will forward samples be interesting?

At least finding an error message test will be easy.  They're all over the place, like this one for neg:

https://github.com/pytorch/pytorch/blob/eec48303c0d847d3a75edbc455f5cd1d7ef098b9/test/test_unary_ufuncs.py#L192

Another candidate op for reviewing what "forward_samples" might look like is addr:

https://github.com/pytorch/pytorch/blob/14fa47631b91f04bca1ffb47efde1cc0bd550f55/test/test_linalg.py#L399

Which seems ideal since its tests are basically input vs reference, and there's no good (or at least no existing good) structure to put them. 

I would be curious to hear your thoughts on what it would take to put a test like the unary ufunc's "test_reference_numerics" into this pattern, too, though.